test/conformance/helpers/broker_data_plane_test_helper.go

```diff
@@ -0,0 +1,458 @@
+				cetest.HasSource(event.Source()),
+				cetest.HasSpecVersion("0.3"),
+			))
+			eventTracker.AssertAtLeast(1, originalEventMatcher)
+		})
+
+		t.Run("Ingress Supports CE1.0", func(t *testing.T) {
+			eventID := "CE1.0"
+			event := ce.NewEvent()
+			event.SetID(eventID)
+			event.SetType(testlib.DefaultEventType)
+			event.SetSource("CE1.0")
+			body := fmt.Sprintf(`{"msg":%q}`, eventID)
+			if err := event.SetData(ce.ApplicationJSON, []byte(body)); err != nil {
+				t.Fatalf("Cannot set the payload of the event: %s", err.Error())
+			}
+
+			client.SendEventToAddressable("v10-test-sender", broker.Name, testlib.BrokerTypeMeta, event)
+			originalEventMatcher := recordevents.MatchEvent(cetest.AllOf(
+				cetest.HasSource(event.Source()),
+			))
+			eventTracker.AssertAtLeast(1, originalEventMatcher)
+
+		})
+		t.Run("Ingress Supports Structured Mode", func(t *testing.T) {
+			eventID := "Structured-Mode"
+			event := ce.NewEvent()
+			event.SetID(eventID)
+			event.SetType(testlib.DefaultEventType)
+			event.SetSource("CEStructured Mode")
+			body := fmt.Sprintf(`{"msg":%q}`, eventID)
+			if err := event.SetData(ce.EncodingStructured.String(), []byte(body)); err != nil {
+				t.Fatalf("Cannot set the payload of the event: %s", err.Error())
+			}
+
+			client.SendEventToAddressable("structured-test-sender", broker.Name, testlib.BrokerTypeMeta, event)
+			originalEventMatcher := recordevents.MatchEvent(cetest.AllOf(
+				cetest.HasSource(event.Source()),
+			))
+			eventTracker.AssertAtLeast(1, originalEventMatcher)
+		})
+
+		t.Run("Ingress Supports Binary Mode", func(t *testing.T) {
+			eventID := "Binary-Mode"
+			event := ce.NewEvent()
+			event.SetID(eventID)
+			event.SetType(testlib.DefaultEventType)
+			event.SetSource("CEBindary Mode")
+			body := fmt.Sprintf(`{"msg":%q}`, eventID)
+			if err := event.SetData(ce.EncodingBinary.String(), []byte(body)); err != nil {
+				t.Fatalf("Cannot set the payload of the event: %s", err.Error())
+			}
+
+			client.SendEventToAddressable("binary-test-sender", broker.Name, testlib.BrokerTypeMeta, event)
+			originalEventMatcher := recordevents.MatchEvent(cetest.AllOf(
+				cetest.HasSource(event.Source()),
+			))
+			eventTracker.AssertAtLeast(1, originalEventMatcher)
+		})
+
+		t.Run("Respond with 2XX on good CE", func(t *testing.T) {
+			eventID := "2hundred-on-good-ce"
+			body := fmt.Sprintf(`{"msg":%q}`, eventID)
+			responseSink := "http://" + client.GetServiceHost(loggerName)
+			client.SendRequestToAddressable("twohundred-test-sender", broker.Name, testlib.BrokerTypeMeta,
+				map[string]string{
+					"ce-specversion": "1.0",
+					"ce-type":        testlib.DefaultEventType,
+					"ce-source":      "2XX on good CE",
+					"ce-id":          eventID,
+					"content-type":   ce.ApplicationJSON,
+				},
+				body,
+				sender.WithResponseSink(responseSink),
+			)
+
+			eventTracker.AssertExact(1, recordevents.MatchEvent(sender.MatchStatusCode(202))) // should probably be a range
```

Comment:
Because you're sending to the same event tracker, you may assert the wrong event "sent event". I think here you should assert together the id of the sent event and the status code (the id of the sender event is the same of the id of the sent event), like:

`MatchEvent(HasId(eventID), sender.MatchStatusCode(202))`